govtech/detectors/cves/cve_2020_3452/src/main/java/com/google/tsunami/plugins/detectors/cves/cve_2020_3452/CVE20203452Detector.java

```diff
@@ -0,0 +1,124 @@
+import static com.google.tsunami.common.net.http.HttpRequest.get;
+import static com.google.common.net.HttpHeaders.CONNECTION;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.protobuf.util.Timestamps;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpHeaders;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import javax.inject.Inject;
+
+/** A {@link VulnDetector} that detects unauthenticated CVE-2020-3452 Cisco ASA/FTD directory traversal. */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "CVE20203452",
+    version = "0.1",
+    description =
+        "This detector checks whether the Cisco ASA/FTD Server is vulnerable to"
+        + "CVE-2020-3452 directory traversal attack.",
+    author = "Eugene Lim (eugene_lim@tech.gov.sg)",
+    bootstrapModule = CVE20203452DetectorBootstrapModule.class)
+public final class CVE20203452Detector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+
+  @Inject
+  public CVE20203452Detector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient).modify().setFollowRedirects(false).build();
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("Starting CVE-2020-3452 detection.");
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + "+CSCOT+/translation-table?"
```

Comment:
Same here, please break this line per 100 column width limit. Thanks!

Reply:
Done!